### PR TITLE
Added support of jumbo frames

### PR DESCRIPTION
**Pull request by ifilippov**
If user wants to have whole JUMBO frame in one mbuf user should
enable MemoryJumbo initialization option. This will lead to extensive
memory usage.

If user wants to have JUMBO frame as chained mbufs user should
enable ChainedJumbo initialization option and then use Next pointers

### CODE REVIEW

**Comment by gshimansky** [comment]
I don't quite like it that to enable jumbo frames it is necessary to modify `low.h` instead of specifying flags to `SystemInit`. It is possible to do this with initialization flags instead?

**Comment by ifilippov** [comment]
It is highly connected with reassembly functionality which is controlled through low.h. If you propose to change this design - it is separate task. For example, I propose to have defines.h file with all user-necessary defines in it - debug, counters, reassembly, jumbo. But it is different task.

File: examples/jumbo.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+// several chained segments. You should enable CHAINED_JUMBO to correctly handle them
+
+// Note: usage of maxPacketSegment > 2014 and also requiredPacketLength > maxPacketSegment
+// is forbidden. If you decicde to chain packets you shouldn't chain non-standart packets
+// Explanation: in this situation you will need to define both
+// MEMORY_JUMBO and CHAINED_JUMBO. This is treated as underfined behaviour
+const requiredPacketLength = 7000
+
+func main() {
+	flow.SystemInit(nil)
```
Comment (gshimansky) [inline examples/jumbo.go]: Shouldn't initialization set necessary jumbo flags here?

File: flow/flow.go [hunk 2]
```diff
@@ -522,6 +522,13 @@ type Config struct {
 	//
 	// If no string is specified, no HTTP server is spawned.
 	StatsHTTPAddress *net.TCPAddr
+	// Enables possibility of IP reassembly via chaining packets
+	ChainedReassembly bool
+	// Enables possibility of handling jumba frames via chaining packets
+	ChainedJumbo bool
```
Comment (gshimansky) [inline flow/flow.go]: jumbo, not jumba

File: examples/jumbo.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	flow.SetStopper(received)
+	flow.SystemStart()
+}
+
+func generate(pkt *packet.Packet, context flow.UserContext) {
+	remain := requiredPacketLength
+	c := 0
+	diff, remain := m(maxPacketSegment, remain)
+	if packet.InitEmptyIPv4Packet(pkt, diff) == false {
+		fmt.Printf("maxPacketSegment %d is more than 2014 bytes without MEMORY_JUMBA defined\n", maxPacketSegment)
```
Comment (gshimansky) [inline examples/jumbo.go]: jumbo, not jumba

File: examples/jumbo.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+import (
+	"fmt"
+	"github.com/intel-go/nff-go/flow"
+	"github.com/intel-go/nff-go/packet"
+	"time"
+)
+
+// You need to define MEMORY_JUMBO in low/low.h for maxPacketSegment more than 2014.
+// Explanation: internal packet buffers are 2048 bytes, first packet will have
```
Comment (gshimansky) [inline examples/jumbo.go]: Shouldn't this comment be phrased differently with initialization flags instead of defines?

**Pull request merged**